test/producePublicFakeFactors.py

```diff
@@ -3,170 +3,156 @@
-                 object='uncertainties_QCD_SS_MVis_Iso_down',
+                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/{CHANNEL}/{CATEGORY}/pieces/uncertainties_QCD_W.root'.format(HOME=home,VERSION=version,CHANNEL=channel,CATEGORY=category),
+                 object='uncertainties_QCD_MVis_Iso_SS2OS_down',
                  vars=['mvis', 'mu_iso']
              ),
-             qcd_ss.find('ff_qcd_ss')
+             qcd_os.find('ff_qcd_os')
          ]
      )
     }
 )
-qcd_os_up = replace_nodes(
+qcd_os_up_stat = replace_nodes(
     qcd_os, 
     {'ff_qcd_os':
      Node(
-         name='ff_qcd_os_up',
-         formula='(1.+{sys_qcd_up})*{ff_qcd_os}',
+         name='ff_qcd_os_up_stat',
+         formula='(1.+{stat_qcd_up})*{ff_qcd_os}',
          leaves=[
              Leaf(
-                 name='sys_qcd_up',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/uncertainties_QCD_W.root'.format(HOME=home,VERSION=version),
-                 object='uncertainties_QCD_MVis_Iso_SS2OS_up',
-                 vars=['mvis', 'mu_iso']
+                 name='stat_qcd_up',
+                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/{CHANNEL}/{CATEGORY}/pieces/FakeFactors_Data_QCD_3D.root'.format(HOME=home,VERSION=version,CHANNEL=channel,CATEGORY=category),
+                 object='FakeFactors_Data_QCDSS_3D_SS_Iso_Medium_SS_InvertIso_Medium_tau_pt_vs_decayMode_error',
+                 vars=['tau_pt','tau_decay','njets']
              ),
              qcd_os.find('ff_qcd_os')
          ]
      )
     }
 )
-qcd_os_down = replace_nodes(
+
+qcd_os_down_stat = replace_nodes(
     qcd_os, 
     {'ff_qcd_os':
      Node(
-         name='ff_qcd_os_down',
-         formula='max(0.,1.-{sys_qcd_down})*{ff_qcd_os}',
+         name='ff_qcd_os_down_stat',
+         formula='max(0.,1.-{stat_qcd_down})*{ff_qcd_os}',
          leaves=[
              Leaf(
-                 name='sys_qcd_down',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/uncertainties_QCD_W.root'.format(HOME=home,VERSION=version),
-                 object='uncertainties_QCD_MVis_Iso_SS2OS_down',
-                 vars=['mvis', 'mu_iso']
+                 name='stat_qcd_down',
+                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/{CHANNEL}/{CATEGORY}/pieces/FakeFactors_Data_QCD_3D.root'.format(HOME=home,VERSION=version,CHANNEL=channel,CATEGORY=category),
+                 object='FakeFactors_Data_QCDSS_3D_SS_Iso_Medium_SS_InvertIso_Medium_tau_pt_vs_decayMode_error',
+                 vars=['tau_pt','tau_decay','njets']
              ),
              qcd_os.find('ff_qcd_os')
          ]
      )
     }
 )
 
+###########################################################################################################
 ### W fake factors
+
 w = Node(
     name='ff_w',
-    formula='{mtcorr_w}*{mviscorr_w}*{ff_raw_w}',
+    formula='{mtcorr_w}*{ff_raw_w}+0*{mviscorr_w}',
```

Comment:
Is the `+0*{mviscorr_w}` what you want? If this is the case it is probably better to just remove the mvis correction.